### PR TITLE
2to3

### PR DESCRIPTION
**Pull request by m-kuhn**

### CODE REVIEW

File: tests/src/python/test_qgsappstartup.py [hunk 1]
```diff
@@ -110,21 +110,21 @@ def testOptionsPath(self):
         if sys.platform[:3] == 'dar':  # Mac
             subdir = 'qgis.org'
         ini = os.path.join(subdir, 'QGIS2.ini')
-        for p in ['test_opts', 'test opts', u'test_optsé€']:
+        for p in ['test_opts', 'test opts', 'test_optsé€']:
```
Comment (sbrunner) [inline tests/src/python/test_qgsappstartup.py]: I don't think that this kind of things will works with Python2.
In Python 3.3, the `u` prefix has been reintroduced. :-)
http://pythonhosted.org/six/#six.u

Comment (sbrunner) [inline tests/src/python/test_qgsappstartup.py]: used `-x unicode` in 2to3?

**Comment by sbrunner** [comment]
To resolve some error I think that you should use http://pythonhosted.org/six/#module-six.moves

**Comment by sbrunner** [comment]
For example this one: https://travis-ci.org/qgis/QGIS/jobs/115255702#L1260

**Comment by m-kuhn** [comment]
Thanks for the feedback.
That was just a quick shot to see what happens and not really thought to be production ready.

**Comment by sbrunner** [comment]
But it's not so bad :-)

**Comment by m-kuhn** [comment]
Reopening this for a discussion about the way to migrate.

There are different options as far as I can see:

Six
===

https://pythonhosted.org/six/

> Six provides simple utilities for wrapping over differences between Python 2 and Python 3. It is intended to support codebases that work on both Python 2 and 3 without modification. six consists of only one Python file, so it is painless to copy into a project.

Python modernize
==============

https://pypi.python.org/pypi/modernize

> A hack on top of 2to3 for modernizing code for hybrid codebases.

Python Future
=============

http://python-future.org/imports.html

They have some comparison to 2to3, six and python modernize in their FAQ

http://python-future.org/faq.html#how-does-this-relate-to-2to3
http://python-future.org/faq.html#what-is-the-relationship-between-future-and-six
http://python-future.org/faq.html#what-is-the-relationship-between-python-future-and-python-modernize

The main difference I can see is that they "only" support Python 3.3+ and 2.6+. But I don't think that would be a criteria for us.

**Comment by sbrunner** [comment]
I mostly see project that use six, but Python Future looks also good :-)

**Comment by luipir** [comment]
I, sometimes, used to code using future to prepare plugins to be used in 2&3 = 2 ;)

**Pull request closed without merging**